### PR TITLE
Update Versions

### PR DESCRIPTION
**Pull request by ezTxmMC**
- updating builds to latest
- adding newest snapshot version 24w18a
- removing potato snapshot
- removing outdated under-versions
- removing buggy versions like 1.20.5

### CODE REVIEW

File: server_versions.json [hunk 1]
```diff
@@ -47,31 +47,27 @@
...
         "version": "1.20.2",
-        "link": "https://api.papermc.io/v2/projects/paper/versions/1.20.2/builds/291/downloads/paper-1.20.2-291.jar"
+        "link": "https://api.papermc.io/v2/projects/paper/versions/1.20.2/builds/318/downloads/paper-1.20.2-318.jar"
       },
       {
         "version": "1.20.1",
         "link": "https://api.papermc.io/v2/projects/paper/versions/1.20.1/builds/196/downloads/paper-1.20.1-196.jar"
       },
-      {
-        "version": "1.20",
```
Comment (FllipEis) [inline server_versions.json:69]: Why did you remove this? I would leave it there

File: server_versions.json [hunk 2]
```diff
@@ -81,50 +77,18 @@
         "version": "1.19.2",
         "link": "https://api.papermc.io/v2/projects/paper/versions/1.19.2/builds/307/downloads/paper-1.19.2-307.jar"
       },
-      {
-        "version": "1.19.1",
```
Comment (FllipEis) [inline server_versions.json:85]: Why did you remove this? I would leave it there

File: server_versions.json [hunk 3]
```diff
@@ -81,50 +77,18 @@
...
-        "link": "https://api.papermc.io/v2/projects/paper/versions/1.19.1/builds/111/downloads/paper-1.19.1-111.jar"
-      },
-      {
-        "version": "1.19",
```
Comment (FllipEis) [inline server_versions.json:89]: Why did you remove this? I would leave it there

File: server_versions.json [hunk 4]
```diff
@@ -81,50 +77,18 @@
...
-        "link": "https://api.papermc.io/v2/projects/paper/versions/1.19/builds/69/downloads/paper-1.19-69.jar"
-      },
       {
         "version": "1.18.2",
-        "link": "https://api.papermc.io/v2/projects/paper/versions/1.18.2/builds/386/downloads/paper-1.18.2-386.jar"
-      },
-      {
-        "version": "1.18.1",
```
Comment (FllipEis) [inline server_versions.json:97]: Why did you remove this? I would leave it there

File: server_versions.json [hunk 5]
```diff
@@ -81,50 +77,18 @@
...
-        "link": "https://papermc.io/api/v2/projects/paper/versions/1.18.1/builds/133/downloads/paper-1.18.1-133.jar"
+        "link": "https://api.papermc.io/v2/projects/paper/versions/1.18.2/builds/388/downloads/paper-1.18.2-388.jar"
       },
       {
         "version": "1.17.1",
         "link": "https://dl.thesimplecloud.eu/download/versions/paper/Paper-1.17.1.jar"
       },
-      {
-        "version": "1.17",
```
Comment (FllipEis) [inline server_versions.json:105]: Why did you remove this? I would leave it there

File: server_versions.json [hunk 6]
```diff
@@ -81,50 +77,18 @@
...
-        "link": "https://dl.thesimplecloud.eu/download/versions/paper/Paper-1.17.jar"
-      },
       {
         "version": "1.16.5",
         "link": "https://dl.thesimplecloud.eu/download/versions/paper/Paper-1.16.5.jar"
       },
-      {
-        "version": "1.16.4",
```
Comment (FllipEis) [inline server_versions.json:113]: Why did you remove this? I would leave it there

File: server_versions.json [hunk 7]
```diff
@@ -81,50 +77,18 @@
...
-        "link": "https://dl.thesimplecloud.eu/download/versions/paper/Paper-1.16.4.jar"
-      },
-      {
-        "version": "1.16.3",
```
Comment (FllipEis) [inline server_versions.json:117]: Why did you remove this? I would leave it there

File: server_versions.json [hunk 8]
```diff
@@ -81,50 +77,18 @@
...
-        "link": "https://dl.thesimplecloud.eu/download/versions/paper/Paper-1.16.3.jar"
-      },
-      {
-        "version": "1.16.2",
```
Comment (FllipEis) [inline server_versions.json:121]: Why did you remove this? I would leave it there

File: server_versions.json [hunk 9]
```diff
@@ -81,50 +77,18 @@
...
-        "link": "https://dl.thesimplecloud.eu/download/versions/paper/Paper-1.16.2.jar"
-      },
-      {
-        "version": "1.16.1",
```
Comment (FllipEis) [inline server_versions.json:125]: Why did you remove this? I would leave it there

File: server_versions.json [hunk 10]
```diff
@@ -182,28 +146,19 @@
...
         "version": "1.20.4",
         "link": "https://piston-data.mojang.com/v1/objects/8dd1a28015f51b1803213892b50b7b4fc76e594d/server.jar"
       },
       {
-        "version": "24w14a",
-        "link": "https://piston-data.mojang.com/v1/objects/960cb0e5c794b02abdbcdbdc15b4de058b222118/server.jar",
-        "isSnapshot": true
-      },
-      {
-        "version": "24w14potato",
```
Comment (FllipEis) [inline server_versions.json:205]: Why did you remove this? I would leave it there

File: server_versions.json
Comment (FllipEis) [inline server_versions.json:1]: Thank you for your pull request!

There are two small things I want to mention:
-  I saw you removed a lot of old versions, but we want every version to be included (including outdated ones)
    - For example, in the future, all snapshots that ever existed will be added
-  We're currently working on automating this process so we don't have to update the versions in this file manually

**Review by FllipEis — CHANGES_REQUESTED** [review]

File: server_versions.json (hunk 2 above)
Comment (ezTxmMC) [reply server_versions.json:85]: The most used versions of the 1.19-Group are 1.19.2 and 1.19.4, the 1.19.1 is outdated and 1.19.2 fixing some bugs of 1.19.1 and lower

File: server_versions.json (hunk 3 above)
Comment (ezTxmMC) [reply server_versions.json:89]: The most used versions of the 1.19-Group are 1.19.2 and 1.19.4, the 1.19 is outdated and 1.19.2 fixing some bugs of 1.19.1 and lower

File: server_versions.json (hunk 5 above)
Comment (ezTxmMC) [reply server_versions.json:105]: 1.17.1 is a big hotfix patch for 1.17

File: server_versions.json (hunk 6 above)
Comment (ezTxmMC) [reply server_versions.json:113]: 1.16.5 is the latest version of the 1.16-Group and has bug fixes, optimizations and balancing

File: server_versions.json (hunk 8 above)
Comment (ezTxmMC) [reply server_versions.json:121]: 1.16.5 is the latest version of the 1.16-Group and has bug fixes, optimizations and balancing

File: server_versions.json (hunk 9 above)
Comment (ezTxmMC) [reply server_versions.json:125]: 1.16.5 is the latest version of the 1.16-Group and has bug fixes, optimizations and balancing,
but this can be leave there, because this is a speedrunner version.

File: server_versions.json [hunk 11]
```diff
@@ -182,28 +146,19 @@
...
-      {
-        "version": "1.20.5",
-        "link": "https://piston-data.mojang.com/v1/objects/79493072f65e17243fd36a699c9a96b4381feb91/server.jar"
-      },
       {
         "version": "1.20.4",
         "link": "https://piston-data.mojang.com/v1/objects/8dd1a28015f51b1803213892b50b7b4fc76e594d/server.jar"
       },
       {
-        "version": "24w14a",
```
[the remark this answers is not in the document]
Comment (ezTxmMC) [reply server_versions.json:200]: i don't remove it, i only replaced it by the latest snapshot version

File: server_versions.json (hunk 10 above)
Comment (ezTxmMC) [reply server_versions.json:205]: april is over and i think its better to remove it, because nobody more plays the april version

File: server_versions.json (hunk 1 above)
Comment (FllipEis) [reply server_versions.json:69]: Yeah, that's true, but there are people using that version and because we're going to use this file also for the v2 this could break the cloud for some people, so we have to leave it there.

That's also applying for all other versions, so I'm not going to reply to every comment

File: server_versions.json
Comment (ezTxmMC) [reply server_versions.json:1]: I think adding all versions is too much, you should only add all latest versions.
If you adding all versions the users don't have a overview over this.
It's already difficult to have a overview with only the latest versions.
Maybe you can separate this with latest by default and an another file for all and the user can select it by the cloud setup or in the config.

Comment (FllipEis) [reply server_versions.json:1]: No, this file is definitely going to have all versions. We think it's better to have everything in one place and it makes it easier for the api.

**Comment by ezTxmMC** [comment]
i have hopefully re-added all versions

**Comment by ezTxmMC** [comment]
@FllipEis

**Comment by ezTxmMC** [comment]
@FllipEis

**Review by FllipEis — APPROVED** [review]

**Comment by FllipEis** [comment]
> @FllipEis

You don't have to tag me, I will reply when I have time 😄
